### PR TITLE
Developed and Tested "Increment decimal coded number"

### PR DESCRIPTION
**Pull request by huangme**
We have already finished and tested this question. During this development, we added a new javascript file called "incNumber.js" for the question body. In addition, we modified codes in "CorrectnessTestObjectFactory.js" and "PythonCodePreprocessorService.js" for handling the case when inputs and outputs are arrays of numbers.

### CODE REVIEW

**Comment by eyurko** [comment]
It looks like this functionality is duplicated in #133. Which of your groups claimed the question first?

**Comment by huangme** [comment]
@eyurko I think my group declared this question first. I put our team members' name (Meng Huang, Yuxin Ouyang) right behind this question in the question document that you sent to us (https://docs.google.com/document/d/12qrxeKZ14_Zf3sSmbVcJYm0REwE8m4aCxbW3m0UhEfk/edit).

**Comment by huangme** [comment]
Hi @eyurko , did you request any change? I saw that you asked me to see your other comment, but I did not find any difference.

**Comment by eyurko** [comment]
Both your group and the group on #133 are claiming that you claimed this question first (they argue that they created the issue first). Resolve it among yourselves and let me know which PR I should review.

File: client/services/code_preprocessors/PythonCodePreprocessorService.js [hunk 1]
```diff
@@ -49,6 +49,8 @@ tie.factory('PythonCodePreprocessorService', [
         return "'" + pythonStringContent + "'";
       } else if (typeof jsonVariable == 'boolean') {
         return jsonVariable ? 'True' : 'False';
+      } else if (typeof jsonVariable === 'number') {
+	      return parseInt(jsonVariable);
```
Comment (seanlip) [inline client/services/code_preprocessors/PythonCodePreprocessorService.js]: Could be a float, right?

Also, please use spaces for indentation, not tabs.

Comment (huangme) [inline client/services/code_preprocessors/PythonCodePreprocessorService.js]: Thanks. I've fixed it.

**Comment by huangme** [comment]
Hi @eyurko @seanlip , I discussed with the group working on the same question as us, and, finally, we decided to merge our branches together. I've already finished the merging step. Please take a look on it. Thank you!

**Comment by huangme** [comment]
@eyurko @seanlip Also, our group should have 4 members right now, but I cannot find one of my teammates when I was trying to add assignees. His name is Yuxin Ouyang @ouyangyuxin .

**Comment by seanlip** [comment]
Ah -- @ouyangyuxin needs to accept the invitation at https://github.com/google/tie/invitations before they'll appear in the list.

Also please could you fix the merge conflicts?

I'll leave the rest of the review to @eyurko.

**Comment by huangme** [comment]
Fixed it!

File: client/domain/CorrectnessTestObjectFactory.js [hunk 2]
```diff
@@ -31,8 +32,14 @@ tie.factory('CorrectnessTestObjectFactory', [
     };
 
     CorrectnessTest.prototype.matchesOutput = function(output) {
-      return this._allowedOutputs.some(function(allowedOutput) {
-        return angular.equals(allowedOutput, output);
+      var arrJson = this._allowedOutputs;
```
Comment (eyurko) [inline client/domain/CorrectnessTestObjectFactory.js]: why isn't this just allowedOutputs?

File: client/domain/CorrectnessTestObjectFactory.js [hunk 3]
```diff
@@ -31,8 +32,14 @@ tie.factory('CorrectnessTestObjectFactory', [
     };
 
     CorrectnessTest.prototype.matchesOutput = function(output) {
-      return this._allowedOutputs.some(function(allowedOutput) {
-        return angular.equals(allowedOutput, output);
+      var allowedOutputs = this._allowedOutputs;
+      var target = output;
+      if (Array.isArray(output)) {
```
Comment (eyurko) [inline client/domain/CorrectnessTestObjectFactory.js]: use Angular.isArray

Comment (seanlip) [inline client/domain/CorrectnessTestObjectFactory.js]: NB: should be angular.isArray (lower case for angular)

**Pull request merged**